### PR TITLE
Add support for Magit-Blob buffers

### PR DESCRIPTION
**Pull request by Miciah**
Use the correct branch or commit when the current buffer is visiting a blob using Magit.

* `git-link.el` (`git-link--commit`): Use the `magit-buffer-revision` variable if the buffer is using Magit-Blob mode.
(`git-link--branch`): Look up the branch using `magit-buffer-revision` and the `magit-rev-branch` function if the buffer is using Magit-Blob mode.
(`git-link--relative-filename`): Use the `magit-buffer-file-name` variable the buffer is using Magit-Blob mode.
(`git-link--using-magit-blob-mode`): New function.
(`git-link`): Set the region and ignore the branch that is checked out into the working tree when using Magit-Blob mode.

### CODE REVIEW

File: git-link.el [hunk 1]
```diff
@@ -233,6 +236,8 @@ As an example, \"gitlab\" will match with both \"gitlab.com\" and
 (defun git-link--branch ()
   (or (git-link--get-config "git-link.branch")
       git-link-default-branch
+      (when (git-link--using-magit-blob-mode)
+        (magit-rev-branch magit-buffer-revision))
```
Comment (sshaw) [inline git-link.el]: Under what magi command does this return a valid branch? I tried with a file opened from a commit via `magit-log` but no branch.

Comment (Miciah) [reply git-link.el]: I open a blob from a branch using `magit-find-file`, and then `(magit-rev-branch magit-buffer-revision)` evaluates to the name of that branch.

You likely opened a blob that is not from the tip of a branch, in which case `magit-buffer-revision` will have the right commit id, but `(magit-rev-branch magit-buffer-revision)` will evaluate to nil.  In this case, I believe the right thing to do would be to return the commit id, but I see that `git-link` is in fact using the checked-out branch.  I will amend my changes to fix this problem.

**Comment by Miciah** [comment]
> I noticed that it does not include the line numbers when linking from a blob.

Thanks for noticing this, I had missed it.  I'll look into it.

**Comment by Miciah** [comment]
I believe I fixed the missing line numbers.  (I'm curious why the `when` form that I modified is needed at all.)

**Comment by sshaw** [comment]
Great, thanks again!

**Comment by sshaw** [comment]
> (I'm curious why the when form that I modified is needed at all.)

Because one can link to a file in a dired buffer or in magit buffers and in these case we don't have an actual file and therefore cannot/should not get the region.

**Pull request merged**